File: Algorithm/10주_완성_알고리즘_코딩테스트_기초/2.알고리즘의_연산량/2E_소수의_판별/2e.js
const fs = require('fs');
const inputData = fs.readFileSync('/dev/stdin').toString().split('\n');


n = Number(inputData[0])

function isPrimeNumber(caseNumber) {
  if (caseNumber <= 1) return 'NO';
  if (caseNumber === 2) return 'YES';
  if (caseNumber % 2 === 0) return 'NO';

  // 3부터 √case까지 홀수만 검사
  for (let i=3; i<(caseNumber ** 0.5)+1; i+=2) {
    if (caseNumber % i === 0) {
      return 'NO'
    }
  }
  return 'YES'
}

for (let i=0; i<n; i++) {
  caseNumber = Number(inputData[i+1]);
  console.log(`Case #${i+1}`);
  console.log(isPrimeNumber(caseNumber));
}

File: Algorithm/10-Week_Coding_Test/2.알고리즘의_연산량/2E_소수의_판별/2e.py
n = int(input())

def isPrimeNumber(case):
    if case <= 1:      # 1 이하는 소수가 아님
        return 'NO'
    if case == 2:      # 2는 소수
        return 'YES'
    if case % 2 == 0:  # 짝수는 소수가 아님
        return 'NO'

    # 3부터 √case까지 홀수만 검사
    for i in range(3, int(case ** 0.5) + 1, 2):
        if case % i == 0:
            return 'NO'
    return 'YES'

for i in range(n):
    case = int(input())
    print(f"Case #{i+1}") 
    print(isPrimeNumber(case))
